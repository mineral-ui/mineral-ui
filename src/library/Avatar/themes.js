/* @flow */
import type { AvatarThemeFn } from './types';

export const avatarTheme: AvatarThemeFn = (baseTheme) => ({
  Avatar_fontSize_small: baseTheme.fontSize_mouse,
  Avatar_fontSize_medium: baseTheme.fontSize_ui,
  Avatar_fontSize_large: baseTheme.h4_fontSize,
  Avatar_fontSize_jumbo: baseTheme.h4_fontSize,
  Avatar_fontWeight: baseTheme.fontWeight_bold,
  Avatar_size_small: baseTheme.size_small,
  Avatar_size_medium: baseTheme.size_medium,
  Avatar_size_large: baseTheme.size_large,
  Avatar_size_jumbo: baseTheme.size_jumbo,

  ...baseTheme
});
