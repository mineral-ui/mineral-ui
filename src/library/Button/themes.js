/* @flow */
import { pxToEm } from '../styles';

import type { ButtonThemeFn } from './types';

// prettier-ignore
export const buttonTheme: ButtonThemeFn = (baseTheme) => ({
  Button_backgroundColor: baseTheme.backgroundColor,
  Button_backgroundColor_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_focus: baseTheme.backgroundColor_focus,
  Button_backgroundColor_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_minimal_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_minimal_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_primary: baseTheme.backgroundColor_themePrimary,
  Button_backgroundColor_primary_active: baseTheme.backgroundColor_themePrimary_active,
  Button_backgroundColor_primary_focus: baseTheme.backgroundColor_themePrimary_focus,
  Button_backgroundColor_primary_hover: baseTheme.backgroundColor_themePrimary_hover,
  Button_borderColor: baseTheme.borderColor,
  Button_borderColor_active: baseTheme.borderColor_theme_active,
  Button_borderColor_focus: baseTheme.borderColor_theme_focus,
  Button_borderColor_hover: baseTheme.borderColor_theme_hover,
  Button_borderRadius: baseTheme.borderRadius_3,
  Button_borderWidth: 1, // px,
  Button_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${baseTheme.borderColor_theme_focus}`,
  Button_color: baseTheme.color_theme,
  Button_color_minimal: baseTheme.color_theme,
  Button_color_primary: baseTheme.color_themePrimary,
  Button_fontWeight: baseTheme.fontWeight_semiBold,
  Button_paddingHorizontal: baseTheme.space_inset_sm_plus,
  Button_paddingIconOnly_small: pxToEm(3),
  Button_paddingIconOnly_medium: pxToEm(7),
  Button_paddingIconOnly_large: pxToEm(7),
  Button_paddingIconOnly_jumbo: pxToEm(13),
  Button_height_small: baseTheme.size_small,
  Button_height_medium: baseTheme.button_size_medium,
  Button_height_large: baseTheme.button_size_large,
  Button_height_jumbo: baseTheme.button_size_jumbo,

  ButtonContent_fontSize: baseTheme.fontSize_ui_sm,
  ButtonContent_fontSize_small: pxToEm(12),

  ButtonIcon_color: baseTheme.icon_color_theme,
  ButtonIcon_margin: baseTheme.space_inset_sm,

  ...baseTheme
});
