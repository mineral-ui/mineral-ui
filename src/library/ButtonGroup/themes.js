/* @flow */
import type { ButtonGroupThemeFn } from './types';

export const buttonGroupTheme: ButtonGroupThemeFn = (baseTheme) => ({
  ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
  ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
  ButtonGroupButton_borderColor_active: baseTheme.borderColor_theme_active,
  ButtonGroupButton_borderColor_hover: baseTheme.borderColor_theme_hover,
  ButtonGroupButton_borderStartColor: baseTheme.borderColor,
  ButtonGroupButton_borderStartColor_checked: 'currentcolor',
  ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,

  ...baseTheme
});
