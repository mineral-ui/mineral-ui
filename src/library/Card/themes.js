/* @flow */
import { pxToEm } from '../styles';

import type {
  CardThemeFn,
  CardActionsThemeFn,
  CardBlockThemeFn,
  CardDividerThemeFn,
  CardFooterThemeFn,
  CardStatusThemeFn,
  CardTitleThemeFn
} from './types';

// prettier-ignore
export const cardTheme: CardThemeFn = (baseTheme) => ({
  Card_backgroundColor: baseTheme.panel_backgroundColor,
  Card_borderColor: baseTheme.panel_borderColor,
  Card_borderRadius: baseTheme.borderRadius_1,
  Card_boxShadow: baseTheme.boxShadow_2,
  Card_boxShadow_focus: `0 0 0 2px ${baseTheme.borderColor_theme_focus}, ${baseTheme.boxShadow_1}`,

  CardRow_marginVertical: baseTheme.space_stack_md,
  CardRow_marginVerticalLast: baseTheme.space_stack_lg,
  CardRow_paddingHorizontal: baseTheme.space_inset_md,

  ...baseTheme
});

export const cardActionsTheme: CardActionsThemeFn = (baseTheme) => ({
  CardActionsAction_spaceInline: baseTheme.space_inline_sm,

  ...baseTheme
});

export const cardBlockTheme: CardBlockThemeFn = (baseTheme) => ({
  CardBlock_fontSize: baseTheme.fontSize_ui,
  CardBlock_lineHeight: baseTheme.lineHeight_prose,

  ...baseTheme
});

export const cardDividerTheme: CardDividerThemeFn = (baseTheme) => ({
  CardDivider_borderColor: baseTheme.borderColor,
  CardDivider_borderWidth: '1px',

  ...baseTheme
});

export const cardFooterTheme: CardFooterThemeFn = (baseTheme) => ({
  CardFooter_backgroundColor: baseTheme.well_backgroundColor,
  CardFooter_borderColor: baseTheme.borderColor,

  CardFooterRow_marginVertical: baseTheme.space_stack_sm,
  CardFooterRow_marginVerticalLast: baseTheme.space_stack_md,

  CardFooterTitle_color: baseTheme.h5_color,
  CardFooterTitle_fontSize: baseTheme.h5_fontSize,
  CardFooterTitle_fontWeight: baseTheme.h5_fontWeight,

  ...baseTheme
});

export const cardStatusTheme: CardStatusThemeFn = (baseTheme) => ({
  CardStatus_fontSize: baseTheme.fontSize_ui,
  CardStatus_fontWeight: baseTheme.fontWeight_regular,

  CardStatusIcon_margin: baseTheme.space_inline_sm,
  CardStatusIcon_size: pxToEm(12),

  ...baseTheme
});

export const cardTitleTheme: CardTitleThemeFn = (baseTheme) => ({
  CardTitle_color: baseTheme.h4_color,
  CardTitle_fontSize: baseTheme.h4_fontSize,
  CardTitle_fontWeight: baseTheme.h4_fontWeight,

  CardTitleAvatar_margin: baseTheme.space_inline_sm,
  CardTitleAvatarSize: baseTheme.size_small,
  CardTitleAvatarSize_large: baseTheme.size_large,

  CardTitleIcon_margin: baseTheme.space_inline_sm,

  CardTitleSecondaryText_color: baseTheme.color_mouse,
  CardTitleSecondaryText_fontSize: baseTheme.fontSize_mouse,
  CardTitleSecondaryText_fontWeight: baseTheme.fontWeight_regular,

  CardSubtitle_color: baseTheme.color_mouse,
  CardSubtitle_fontSize: baseTheme.fontSize_mouse,
  CardSubtitle_fontWeight: baseTheme.fontWeight_regular,
  CardSubtitle_marginTop: baseTheme.space_stack_sm,

  ...baseTheme
});
