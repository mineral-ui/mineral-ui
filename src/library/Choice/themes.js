/* @flow */
import { pxToEm } from '../styles';

import type { ChoiceThemeFn, ChoiceGroupThemeFn } from './types';

export const choiceTheme: ChoiceThemeFn = (baseTheme) => {
  const colors = {
    background: baseTheme.input_backgroundColor,
    regular: baseTheme.borderColor,
    checked: baseTheme.borderColor_theme,
    focus: baseTheme.borderColor_theme_focus,
    hover: baseTheme.borderColor_theme_hover
  };

  // prettier-ignore
  return {
    ChoiceControl_backgroundColor: colors.background,
    ChoiceControl_backgroundColor_checked: colors.checked,
    ChoiceControl_backgroundColor_checkedHover: colors.hover,
    ChoiceControl_borderColor: colors.regular,
    ChoiceControl_borderColor_hover: colors.hover,
    ChoiceControl_borderColor_checked: colors.checked,
    ChoiceControl_borderColor_checkedHover: colors.hover,
    ChoiceControl_borderRadius: baseTheme.borderRadius_1,
    ChoiceControl_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${colors.focus}`,
    ChoiceControl_size: pxToEm(16),
    ChoiceControl_size_jumbo: pxToEm(24),

    ChoiceText_color: baseTheme.color,
    ChoiceText_fontSize: baseTheme.fontSize_ui,
    ChoiceText_fontSize_small: pxToEm(12),
    ChoiceText_marginHorizontal: baseTheme.space_inline_md,

    ...baseTheme
  };
};

export const choiceGroupTheme: ChoiceGroupThemeFn = (baseTheme) => ({
  ChoiceGroupControl_marginHorizontal_inline: baseTheme.space_inline_xl,
  ChoiceGroupControl_marginVertical_stacked: baseTheme.space_stack_md,
  ChoiceGroupControl_marginVertical_stackedJumbo: baseTheme.space_stack_lg,

  ...baseTheme
});
