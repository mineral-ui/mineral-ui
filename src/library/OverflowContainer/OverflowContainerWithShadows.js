/* @flow */
import React, { Component } from 'react';
import debounce from 'lodash.debounce';
import { createStyledComponent } from '../styles';
import { createThemedComponent, mapComponentThemes } from '../themes';
import _OverflowContainer, {
  componentTheme as overflowContainerComponentTheme
} from './OverflowContainer';

type Props = {
  children?: React$Node,
  hideScrollbars?: boolean,
  scrollX?: boolean,
  scrollY?: boolean
};

type State = {
  hasShadowBottom: boolean,
  hasShadowLeft: boolean,
  hasShadowRight: boolean,
  hasShadowTop: boolean
};

// prettier-ignore
export const componentTheme = (baseTheme: Object) =>
  mapComponentThemes(
    {
      name: 'OverflowContainer',
      theme: overflowContainerComponentTheme(baseTheme)
    },
    {
      name: 'OverflowContainerWithShadows',
      theme: {
        OverflowContainerWithShadows_boxShadowBottom: `inset 0 -8px 8px -8px ${baseTheme.color_gray_60}, inset 0 -1px ${baseTheme.borderColor}`,
        OverflowContainerWithShadows_boxShadowLeft: `inset 8px 0 8px -8px ${baseTheme.color_gray_60}, inset 1px 0 ${baseTheme.borderColor}`,
        OverflowContainerWithShadows_boxShadowRight: `inset -8px 0 8px -8px ${baseTheme.color_gray_60}, inset -1px 0 ${baseTheme.borderColor}`,
        OverflowContainerWithShadows_boxShadowTop: `inset 0 8px 8px -8px ${baseTheme.color_gray_60}, inset 0 1px ${baseTheme.borderColor}`,
      }
    },
    baseTheme
  );

const ThemedOverflowContainer = createThemedComponent(
  _OverflowContainer,
  ({ theme: baseTheme }) =>
    mapComponentThemes(
      {
        name: 'OverflowContainerWithShadows',
        theme: componentTheme(baseTheme)
      },
      {
        name: 'OverflowContainer',
        theme: {}
      },
      baseTheme
    )
);

const getBoxShadows = ({ theme: baseTheme, ...restProps }) => {
  const theme = componentTheme(baseTheme);
  const boxShadows = Object.keys(restProps)
    .filter((prop) => prop.indexOf('hasShadow') !== -1)
    .reduce((acc, prop) => {
      if (restProps[prop]) {
        const edge = prop.split('hasShadow')[1];
        acc.push(theme[`OverflowContainerWithShadows_boxShadow${edge}`]);
      }
      return acc;
    }, []);
  return boxShadows.length ? boxShadows.join(',') : undefined;
};

// [1] - 1px to avoid unwanted scrollbar,
//       1px in to avoid potentially cutting off of focus ring of
//       subcomponents in body
const styles = {
  scroller: ({ scrollX, scrollY }) => ({
    flex: '1 1 auto',

    ...(scrollX
      ? {
          overflowY: 'hidden',
          paddingLeft: 2, // [1]
          paddingRight: 2 // [1]
        }
      : undefined),

    ...(scrollY
      ? {
          overflowX: 'hidden',
          paddingBottom: 2, // [1]
          paddingTop: 2 // [1]
        }
      : undefined)
  }),
  root: (props) => {
    const boxShadow = getBoxShadows(props);
    return boxShadow
      ? {
          display: 'flex',
          position: 'relative',

          '&::before': {
            bottom: 0,
            boxShadow,
            content: '""',
            pointerEvents: 'none',
            position: 'absolute',
            left: 0,
            right: 0,
            top: 0
          }
        }
      : undefined;
  }
};

const Root = createStyledComponent('div', styles.root, {
  displayName: 'OverflowContainerWithShadows'
});
const Scroller = createStyledComponent(
  ThemedOverflowContainer,
  styles.scroller,
  {
    displayName: 'Scroller',
    forwardProps: ['containerRef']
  }
);

/**
 * OverflowContainerWithShadows
 */
export default class OverflowContainerWithShadows extends Component<
  Props,
  State
> {
  container: ?HTMLElement;

  setContainerRef = (node: HTMLElement) => {
    this.container = node;
  };

  componentDidMount() {
    this.applyShadows(this.container);
  }

  render() {
    const {
      children,
      hideScrollbars,
      scrollX,
      scrollY,
      ...restProps
    } = this.props;
    const isScrollable = Boolean(
      this.state &&
        Object.keys(this.state).filter((key) => this.state[key]).length
    );
    const rootProps = {
      ...this.state,
      ...restProps
    };
    const scrollerProps = {
      hideScrollbars,
      containerRef: this.setContainerRef,
      onScroll: (scrollX || scrollY) && this.handleScroll,
      scrollX,
      scrollY,
      // Set tabIndex when scrollable so user can scroll with keyboard
      tabIndex: isScrollable ? 0 : undefined
    };

    return (
      <Root {...rootProps}>
        <Scroller {...scrollerProps}>{children}</Scroller>
      </Root>
    );
  }

  handleScroll = (event: SyntheticEvent<HTMLElement>) => {
    this.applyShadows(event.currentTarget);
  };

  applyShadows = debounce(
    (element: HTMLElement) => {
      const { scrollX, scrollY } = this.props;
      if (scrollX) {
        this.setState({
          hasShadowLeft: element.scrollLeft > 0,
          hasShadowRight:
            element.scrollLeft + element.clientWidth < element.scrollWidth
        });
      }
      if (scrollY) {
        this.setState({
          hasShadowTop: element.scrollTop > 0,
          hasShadowBottom:
            element.scrollTop + element.clientHeight < element.scrollHeight
        });
      }
    },
    100,
    { leading: true }
  );
}
