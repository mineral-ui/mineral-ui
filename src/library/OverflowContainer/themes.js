/* @flow */
import { mapComponentThemes } from '../themes';

import type {
  OverflowContainerThemeFn,
  OverflowContainerWithShadowsThemeFn
} from './types';

// prettier-ignore
export const overflowContainerTheme: OverflowContainerThemeFn = (baseTheme) => ({
  OverflowContainer_outline_focus: `1px solid ${baseTheme.borderColor_theme_focus}`,

  ...baseTheme
});

// prettier-ignore
// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const overflowContainerWithShadowsTheme: OverflowContainerWithShadowsThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'OverflowContainer',
      theme: overflowContainerTheme(baseTheme)
    },
    {
      name: 'OverflowContainerWithShadows',
      theme: {
        OverflowContainerWithShadows_boxShadowBottom: `inset 0 -8px 8px -8px ${baseTheme.color_gray_60}, inset 0 -1px ${baseTheme.borderColor}`,
        OverflowContainerWithShadows_boxShadowLeft: `inset 8px 0 8px -8px ${baseTheme.color_gray_60}, inset 1px 0 ${baseTheme.borderColor}`,
        OverflowContainerWithShadows_boxShadowRight: `inset -8px 0 8px -8px ${baseTheme.color_gray_60}, inset -1px 0 ${baseTheme.borderColor}`,
        OverflowContainerWithShadows_boxShadowTop: `inset 0 8px 8px -8px ${baseTheme.color_gray_60}, inset 0 1px ${baseTheme.borderColor}`,
      }
    },
    baseTheme
  );
