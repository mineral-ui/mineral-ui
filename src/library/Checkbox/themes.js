/* @flow */
import { mapComponentThemes } from '../themes';
import { choiceTheme, choiceGroupTheme } from '../Choice';

import type { CheckboxThemeFn, CheckboxGroupThemeFn } from './types';

import { pxToEm } from '../styles';

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxTheme: CheckboxThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'Choice',
      theme: choiceTheme(baseTheme)
    },
    {
      name: 'Checkbox',
      theme: {
        CheckboxControl_backgroundColor_checked: baseTheme.checkbox_backgroundColor_checked,
        CheckboxControl_backgroundColor_checkedHover: baseTheme.checkbox_backgroundColor_checkedHover,
        CheckboxControl_borderColor: baseTheme.checkbox_borderColor,
        CheckboxControl_borderColor_checked: baseTheme.checkbox_borderColor_checked,
        CheckboxControl_borderColor_checkedHover: baseTheme.checkbox_borderColor_checkedHover,
        CheckboxControl_borderColor_hover: baseTheme.checkbox_borderColor_hover,
        CheckboxControl_borderRadius: baseTheme.borderRadius_2,
        CheckboxControl_borderWidth: `${baseTheme.checkbox_borderWidth}px`,
        CheckboxControl_size: pxToEm(20),
        CheckboxText_marginHorizontal: baseTheme.space_inline_sm
      }
    },
    baseTheme
  );

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxGroupTheme: CheckboxGroupThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'ChoiceGroup',
      theme: choiceGroupTheme(baseTheme)
    },
    {
      name: 'CheckboxGroup',
      theme: {}
    },
    baseTheme
  );
