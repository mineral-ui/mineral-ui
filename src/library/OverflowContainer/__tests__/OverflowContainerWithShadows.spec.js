/* @flow */
import React from 'react';
import { mineralTheme } from '../../themes';
import { getProcessedComponentThemeKeys } from '../../themes/processComponentTheme';
import OverflowContainerWithShadows, {
  componentTheme
} from '../OverflowContainerWithShadows';
import { mountInThemeProvider } from '../../../../utils/enzymeUtils';
import testThemeOverrides from '../../../../utils/testThemeOverrides';

/*
 * [1] This getter does not exist on the HTMLElement.prototype in JSDOM, so we
 *     must mock it on the global. `clientWidth` and `offsetLeft`, also used by
 *     OverflowContainer always return 0, which is conveniently useful.
 * [2] Then we must clean up our global mocking in [1]
 */

const defaultProps = {
  scrollX: true
};

const mountOverflowContainerWithShadows = (props = {}) => {
  const containerProps = {
    ...defaultProps,
    ...props
  };
  return mountInThemeProvider(
    <OverflowContainerWithShadows {...containerProps} />
  );
};

describe('OverflowContainerWithShadows', () => {
  it('renders', () => {
    const [, overflowContainer] = mountOverflowContainerWithShadows();

    expect(overflowContainer.exists()).toEqual(true);
  });

  describe('theme overrides', () => {
    testThemeOverrides(
      <OverflowContainerWithShadows {...defaultProps} />,
      getProcessedComponentThemeKeys(componentTheme(mineralTheme), {
        excludeKeys: [
          'OverflowContainerWithShadows_boxShadowBottom',
          'OverflowContainerWithShadows_boxShadowLeft',
          'OverflowContainerWithShadows_boxShadowRight',
          'OverflowContainerWithShadows_boxShadowTop'
        ]
      })
    );
  });

  it('does not apply shadow when not scrollable', () => {
    const [, overflowContainer] = mountInThemeProvider(
      <OverflowContainerWithShadows {...defaultProps} />
    );

    expect(overflowContainer).toMatchSnapshot();
  });

  it('applies shadow when scrollable', () => {
    // [1]
    // $FlowFixMe
    Object.defineProperties(window.HTMLElement.prototype, {
      scrollWidth: {
        // Note that this mocked getter is designed to always trigger overflow
        // because it's greater than offsetLeft + offsetWidth
        get: () => 10,
        configurable: true
      }
    });

    const [, overflowContainer] = mountInThemeProvider(
      <OverflowContainerWithShadows {...defaultProps} />
    );

    // [2]
    // $FlowFixMe
    Object.defineProperties(window.HTMLElement.prototype, {
      scrollWidth: {
        get: undefined
      }
    });

    expect(overflowContainer).toMatchSnapshot();
  });
});
