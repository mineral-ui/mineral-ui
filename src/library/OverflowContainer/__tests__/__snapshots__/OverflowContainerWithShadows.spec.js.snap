// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverflowContainerWithShadows applies shadow when scrollable 1`] = `
.emotion-1 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: hidden;
  padding-left: 2px;
  padding-right: 2px;
}

.emotion-0 {
  outline: 0;
  overflow-x: auto;
  -webkit-transition: outline 0.1s 0.25s;
  transition: outline 0.1s 0.25s;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: hidden;
  padding-left: 2px;
  padding-right: 2px;
}

.emotion-0:focus {
  outline: 1px solid #1d5bbf;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.emotion-5::before {
  bottom: 0;
  box-shadow: inset -8px 0 8px -8px #8e99ab,inset -1px 0 #c8d1e0;
  content: "";
  pointer-events: none;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
}

<OverflowContainerWithShadows
  scrollX={true}
>
  <OverflowContainerWithShadows
    hasShadowLeft={false}
    hasShadowRight={true}
  >
    <div
      className="emotion-5"
    >
      <Scroller
        containerRef={[Function]}
        onScroll={[Function]}
        scrollX={true}
        tabIndex={0}
      >
        <Scroller
          className="emotion-1"
          containerRef={[Function]}
          onScroll={[Function]}
          scrollX={true}
          tabIndex={0}
        >
          <Themed(OverflowContainer)
            className="emotion-1"
            containerRef={[Function]}
            onScroll={[Function]}
            scrollX={true}
            tabIndex={0}
          >
            <ThemeProvider>
              <ThemeProvider>
                <OverflowContainer
                  className="emotion-1"
                  containerRef={[Function]}
                  onScroll={[Function]}
                  scrollX={true}
                  tabIndex={0}
                >
                  <OverflowContainer
                    className="emotion-1"
                    containerRef={[Function]}
                    innerRef={[Function]}
                    onScroll={[Function]}
                    scrollX={true}
                    tabIndex={0}
                  >
                    <div
                      className="emotion-0"
                      onScroll={[Function]}
                      tabIndex={0}
                    >
                      <EventListener
                        listeners={
                          Array [
                            Object {
                              "event": "resize",
                              "handler": [Function],
                              "target": "window",
                            },
                          ]
                        }
                      />
                    </div>
                  </OverflowContainer>
                </OverflowContainer>
              </ThemeProvider>
            </ThemeProvider>
          </Themed(OverflowContainer)>
        </Scroller>
      </Scroller>
    </div>
  </OverflowContainerWithShadows>
</OverflowContainerWithShadows>
`;

exports[`OverflowContainerWithShadows does not apply shadow when not scrollable 1`] = `
.emotion-1 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: hidden;
  padding-left: 2px;
  padding-right: 2px;
}

.emotion-0 {
  outline: 0;
  overflow-x: auto;
  -webkit-transition: outline 0.1s 0.25s;
  transition: outline 0.1s 0.25s;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: hidden;
  padding-left: 2px;
  padding-right: 2px;
}

.emotion-0:focus {
  outline: 1px solid #1d5bbf;
}

<OverflowContainerWithShadows
  scrollX={true}
>
  <OverflowContainerWithShadows
    hasShadowLeft={false}
    hasShadowRight={false}
  >
    <div
      className="emotion-5"
    >
      <Scroller
        containerRef={[Function]}
        onScroll={[Function]}
        scrollX={true}
      >
        <Scroller
          className="emotion-1"
          containerRef={[Function]}
          onScroll={[Function]}
          scrollX={true}
        >
          <Themed(OverflowContainer)
            className="emotion-1"
            containerRef={[Function]}
            onScroll={[Function]}
            scrollX={true}
          >
            <ThemeProvider>
              <ThemeProvider>
                <OverflowContainer
                  className="emotion-1"
                  containerRef={[Function]}
                  onScroll={[Function]}
                  scrollX={true}
                >
                  <OverflowContainer
                    className="emotion-1"
                    containerRef={[Function]}
                    innerRef={[Function]}
                    onScroll={[Function]}
                    scrollX={true}
                  >
                    <div
                      className="emotion-0"
                      onScroll={[Function]}
                    >
                      <EventListener
                        listeners={
                          Array [
                            Object {
                              "event": "resize",
                              "handler": [Function],
                              "target": "window",
                            },
                          ]
                        }
                      />
                    </div>
                  </OverflowContainer>
                </OverflowContainer>
              </ThemeProvider>
            </ThemeProvider>
          </Themed(OverflowContainer)>
        </Scroller>
      </Scroller>
    </div>
  </OverflowContainerWithShadows>
</OverflowContainerWithShadows>
`;
